ratpack-core/src/main/java/ratpack/http/client/internal/DefaultHttpClient.java

```diff
@@ -49,9 +51,13 @@
   private final HttpChannelPoolMap channelPoolMap = new HttpChannelPoolMap() {
     @Override
     protected ChannelPool newPool(HttpChannelKey key) {
+
       Bootstrap bootstrap = new Bootstrap()
         .remoteAddress(key.host, key.port)
         .group(key.execution.getEventLoop())
+        .resolver(key.execution.maybeGet(AddressResolverGroup.class)
```

Comment:
This is the same pattern we do over in `raptack-retrofit2`, but it's ugly and still requires that users explicitly bind or provide an `ExecInitializer` that address `AddressResolverGroup` to forked executions.

Reply:
I changed this pattern, so that we looking in the execution and if nothing is there, we get the default netty backed resolver (which defaults its configuration to the system properties in `/etc/resolv.conf`). This means a user will still need to explicitly ensure that `AddressResolverGroup` is registered in any forked executions (including `ParallelBatch`) if they do any customizations.

Also, not sure what the impact of creating/gc'ing those would be....it doesn't look like there's any state really.

One oddity here, is that we are finding the `AddressResolverGroup` from the `Execution`, if you are using Guice and injecting `HttpClient`, the `AddressResolverGroup` doesn't come along with it. It's almost as though `HttpClient` should also inject `AddressResolverGroup`, but the Ratpack `Registry` doesn't do dependency binding.